docs/DeveloperGuide.adoc

```diff
@@ -344,85 +344,61 @@ down the list of people.
 ** Cons: There may be 2 people with the same name and thus result in an error.
 
 === Undo / Redo feature
-==== Proposed Implementation
+The undo command enables users to undo their previous commands while the redo feature enables users to redo their undone commands.
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
-Additionally, it implements the following operations:
+==== Undo Implementation
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+The undo feature is facilitated by the HistoryManager. The commands stack and the addressBooks stack in the HistoryManager class are used to govern the undo feature.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+The static commands stack keeps track of all the commands that have been executed by the user while the static addressBooks stack keeps track of the corresponding state of the address book following the executed command.
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+Given below is an example usage scenario and how the undo mechanism behaves at each step.
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+Step 1. The user launches the application for the first time. The HistoryManager will be initialised with the initial address book state pushed to the addressBook stack.
 
-image::UndoRedoState0.png[]
+image::initialStack.png[width=450]
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Step 2. The user executes the delete 3 command to delete the 3rd person in the Athletick list. The delete command will be pushed into the commands stack in and the new state of the address book (after executing the delete command) will be pushed to the addressBooks stack as well.
```

Comment:
Is there a reason for `addressBook` and `addressBooks` being used? There is some inconsistency here.